File: articles/the-philosophy-of-algorithms/go/delegation_risk.go
package main

import "fmt"

func delegationRisk(decisionSeverity, automationLevel, opacity float64) float64 { risk := decisionSeverity * automationLevel * opacity; if risk < 0 { return 0 }; if risk > 1 { return 1 }; return risk }
func main() { fmt.Printf("%.6f\n", delegationRisk(0.95, 0.95, 0.80)) }

File: articles/the-philosophy-of-algorithms/rust/delegation_risk.rs
fn delegation_risk(decision_severity: f64, automation_level: f64, opacity: f64) -> f64 {
    (decision_severity * automation_level * opacity).clamp(0.0, 1.0)
}

fn main() { println!("{:.6}", delegation_risk(0.95, 0.95, 0.80)); }
